Sources/MongoSwift/BSON/BSONDecoder.swift

```diff
@@ -192,25 +192,25 @@ internal struct _BSONDecodingStorage {
 /// Extend _BSONDecoder to add methods for "unboxing" values as various types.
 extension _BSONDecoder {
 
-    fileprivate func unboxBSONValue<T: BSONValue>(_ value: BSONValue?, as type: T.Type) throws -> T? {
+    fileprivate func unboxBSONValue<T: BSONValue>(_ value: BSONValue, as type: T.Type) throws -> T {
         guard let typed = value as? T else {
             throw DecodingError._typeMismatch(at: self.codingPath, expectation: type, reality: value)
         }
         return typed
     }
 
-    fileprivate func unboxNumber<T: CodableNumber>(_ value: BSONValue?, as type: T.Type) throws -> T? {
-        guard let unwrapped = value else {
+    fileprivate func unboxNumber<T: CodableNumber>(_ value: BSONValue, as type: T.Type) throws -> T {
+        guard !(value is NSNull) else {
```

Comment:
Ditto my comment to your earlier comment for `_BSONDecoder.container()`.

I believe that when we reach a consensus on `_BSONDecoder.container()`, we can propagate it to other instances.

Reply:
I just realized that `_typeMismatch` and `_numberMismatch` are both just convenience methods to create a `DecodingError.typeMismatch`. This may actually be unrelated to my `_BSONDecoder.container()` question above, since that is about two different error codes and JSONDecoder's prior art.

Reply:
ya, these could probably also just be lumped under the `numberMismatch` error, the errors thrown are almost identical so may as well simplify the logic.

Reply:
Done.